cluster/cluster_test.go

```diff
@@ -75,20 +87,201 @@ var _ = Describe("cluster/cluster tests", func() {
+			result, err := cluster.GetSegmentConfigurationFromFile("/tmp/")
+
+			// Assertions
+			Expect(err).To(HaveOccurred())
+			Expect(result).To(BeNil())
+			Expect(err.Error()).To(ContainSubstring("Failed to open file /tmp/gpsegconfig_dump. Error: open /tmp/gpsegconfig_dump: no such file or directory"))
+		})
+
+		It("should return an error for a file with less than 9 number of fields", func() {
+			// Create a temporary file with incorrect fields content
+			content := "invalid_content\n"
+			tempConfFile := createSegConfigFile(content)
+
+			// Call the function under test
+			result, err := cluster.GetSegmentConfigurationFromFile(os.TempDir())
+
+			// Assertions
+			Expect(err).To(HaveOccurred())
+			Expect(result).To(BeNil())
+			Expect(err.Error()).To(ContainSubstring("Unexpected number of fields (1) in line: invalid_content"))
+
+			// Cleanup
+			os.Remove(tempConfFile.Name())
+		})
+
+		It("should return an error for a file with more than 10 number of fields", func() {
+			// Create a temporary file with incorrect fields content
+			content := "1 -1 p p n u 7000 localhost localhost /data/dir-1 dummy\n"
+			tempConfFile := createSegConfigFile(content)
+
+			// Call the function under test
+			result, err := cluster.GetSegmentConfigurationFromFile(os.TempDir())
+
+			// Assertions
+			Expect(err).To(HaveOccurred())
+			Expect(result).To(BeNil())
+			Expect(err.Error()).To(ContainSubstring("Unexpected number of fields (11) in line: 1 -1 p p n u 7000 localhost localhost /data/dir-1 dummy"))
+
+			// Cleanup
+			os.Remove(tempConfFile.Name())
+		})
+
+		It("should fail when there is type conversion error", func() {
+			// Create a temporary file with one invalid int field
+			content := "1a -1 p p n u 7000 localhost localhost /data/dir1\n"
+			tempConfFile := createSegConfigFile(content)
+
+			//Call the function under test
+			result, err := cluster.GetSegmentConfigurationFromFile(os.TempDir())
+
+			// Assertions
+			Expect(err).To(HaveOccurred())
+			Expect(result).To(BeNil())
+			Expect(err.Error()).To(ContainSubstring("Failed to convert dbID with value 1a to an int. Error: strconv.Atoi: parsing \"1a\": invalid syntax"))
+
+			//Cleanup
+			os.Remove(tempConfFile.Name())
+		})
+
+	})
 	Describe("GetSegmentConfiguration", func() {
-		header := []string{"contentid", "hostname", "datadir"}
-		localSegOne := []driver.Value{"0", "localhost", "/data/gpseg0"}
-		localSegTwo := []driver.Value{"1", "localhost", "/data/gpseg1"}
-		remoteSegOne := []driver.Value{"2", "remotehost", "/data/gpseg2"}
+		header := []string{"dbid","contentid","role","preferredrole", "mode", "status","port","hostname","address", "datadir"}
+		localSegOne := []driver.Value{1,0,"p","p","s","u", 6002,"localhost","127.0.0.1", "/data/gpseg0"}
+		localSegTwo := []driver.Value{2,1,"m","m","s","u", 6003,"localhost","127.0.0.1", "/data/gpseg1"}
+		remoteSegOne := []driver.Value{3,2,"p","m","s","u",6004, "remotehost","127.0.0.1", "/data/gpseg2"}
 
```

Comment:
This looks good to me, but before committing, can we run the `go fmt` for this file?